### PR TITLE
feat(Calendar/RangeCalendar): pagination function props

### PR DESCRIPTION
**Pull request by epr3**
Hello,

This PR adds new `nextPage`/`prevPage` function properties to both `CalendarRoot` and `CalendarNext`/`CalendarPrev` components (their Range and Picker counterparts included) which enables more control in terms of handling pagination inside the calendar, while also enabling a more flexible usage of the next/prev components which have all the data attributes and keyboard interactions as opposed to the regular HTML `button`.

The `CalendarNext`/`CalendarPrev` function properties overwrite the function properties defined on `CalendarRoot` (this addresses the use case where you would want pagination by year along with pagination by month).

It should help address the use case described in #966 .

It also deprecates the `step` property on `CalendarNext`/`CalendarPrev`, which could be removed in `v2`.

I am looking forward to any feedback and/or improvement suggestions.

Thanks! 🚀

### CODE REVIEW

File: packages/radix-vue/src/Calendar/CalendarPrev.vue [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import type { DateValue } from '@internationalized/date'
 
 export interface CalendarPrevProps extends PrimitiveProps {
-/** The calendar unit to go back */
+/** The calendar unit to go back 
+* @deprecated Use `prevPage` instead
+*/
   step?: CalendarIncrement
+/** The function to be used for the prev page. Overwrites the `prevPage` function set on the `CalendarRoot`. */
+  prevPage?: (placeholder: DateValue) => DateValue
```
Comment (zernonia) [inline packages/radix-vue/src/Calendar/CalendarPrev.vue:12]: why do we need `prevPage` if we already provide the props in `CalendarRoot`?

File: packages/radix-vue/src/Calendar/CalendarNext.vue [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import type { DateValue } from '@internationalized/date'
 
 export interface CalendarNextProps extends PrimitiveProps {
-/** The calendar unit to go forward */
+/** The calendar unit to go forward 
+* @deprecated Use `nextPage` instead
+*/
   step?: CalendarIncrement
+/** The function to be used for the next page. Overwrites the `nextPage` function set on the `CalendarRoot`. */
+  nextPage?: (placeholder: DateValue) => DateValue
```
Comment (zernonia) [inline packages/radix-vue/src/Calendar/CalendarNext.vue:12]: why do we need `nextPage` if we already provide the props in `CalendarRoot`?

File: packages/radix-vue/src/Calendar/CalendarRoot.vue [hunk 3]
```diff
@@ -81,6 +81,10 @@ interface BaseCalendarRootProps extends PrimitiveProps {
   isDateUnavailable?: Matcher
   /** The reading direction of the calendar when applicable. <br> If omitted, inherits globally from `ConfigProvider` or assumes LTR (left-to-right) reading mode. */
   dir?: Direction
+  /** A function that returns the next page of the calendar. It receives the current placeholder as an argument inside the component. */
+  nextPage?: (placeholder: DateValue) => DateValue
+  /** A function that returns the previous page of the calendar. It receives the current placeholder as an argument inside the component. */
+  prevPage?: (placeholder: DateValue) => DateValue
```
Comment (zernonia) [inline packages/radix-vue/src/Calendar/CalendarRoot.vue:87]: we we have the default function to behave like `step:  'month'`? So dev dones't need to provide the props everytime

File: packages/radix-vue/src/Calendar/CalendarRoot.vue [hunk 4]
```diff
@@ -169,6 +173,8 @@ const {
   isDateUnavailable: propsIsDateUnavailable,
   calendarLabel,
   defaultValue,
+  nextPage: propsNextPage,
+  prevPage: propsPrevPage,
```
Comment (zernonia) [inline packages/radix-vue/src/Calendar/CalendarRoot.vue:177]: why declare the name with `propsXXX` if there's no naming conflict? 🤔

File: packages/radix-vue/src/Calendar/CalendarPrev.vue (hunk 1 above)
Comment (epr3) [reply packages/radix-vue/src/Calendar/CalendarPrev.vue:12]: For the case when you want multiple `CalendarNext`/`CalendarPrev` components inside the `CalendarRoot` (e.g. the default `CalendarNext`/`CalendarPrev` increments by `month`, the others by `year`) like in the `CalendarWithYearIncrements` story.

File: packages/radix-vue/src/Calendar/CalendarNext.vue (hunk 2 above)
Comment (epr3) [reply packages/radix-vue/src/Calendar/CalendarNext.vue:12]: Same explanation as for `CalendarPrev`.

File: packages/radix-vue/src/Calendar/CalendarRoot.vue (hunk 3 above)
Comment (epr3) [reply packages/radix-vue/src/Calendar/CalendarRoot.vue:87]: Yes, the default is like `step: 'month'`.

File: packages/radix-vue/src/Calendar/CalendarRoot.vue (hunk 4 above)
Comment (epr3) [reply packages/radix-vue/src/Calendar/CalendarRoot.vue:177]: ESLint said I had a naming conflict with this line: https://github.com/radix-vue/radix-vue/blob/6ca465cc92140c142404f42559d41c1a0c6284b2/packages/radix-vue/src/Calendar/CalendarRoot.vue#L213 (cannot redeclare block-scoped variable 'nextPage')

**Review by zernonia — APPROVED** [review]
LGTM! Nicely done @epr3 !
Will be release in `1.9`! 😁

**Pull request merged**